plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,305 @@
+				{float64(30.33), int64(123456789), int64(math.MaxInt64), "Made in China.", bool(false), "123456789011"},
+			},
+			DataTypesList: [][]client.TSDataType{
+				{client.DOUBLE, client.INT64},
+				{client.DOUBLE, client.INT64},
+				{client.DOUBLE, client.INT64, client.INT64, client.TEXT, client.BOOLEAN, client.TEXT},
+			},
+			TimestampList: []int64{
+				constTestTimestamp.UnixNano(), constTestTimestamp.UnixNano(), constTestTimestamp.UnixNano(),
+			},
+		}
+	)
+	cli001 := newTestClient()
+	cli002 := newTestClient()
+	cli002.ConvertUint64To = "text"
+	cli003 := newTestClient()
+	cli003.TimeStampUnit = "second"
+	cli004 := newTestClient()
+	cli004.TreatTagsAs = "fields"
+	cli005 := newTestClient()
+	cli005.TreatTagsAs = "device_id"
+	tests := []struct {
+		name        string
+		cli         *IoTDB
+		expectedRWT recordsWithTags
+		metrics     []telegraf.Metric
+		needModify  bool // if true, modify before comparing; if false, do not call `modifyRecordsWithTags`
+	}{
+		{
+			name:        "01 test normal config",
+			cli:         cli001,
+			expectedRWT: testRecordsWithTags001,
+			metrics:     testMetrics001,
+			needModify:  false,
+		},
+		{
+			name:        "02 test convert unsigned int to text",
+			cli:         cli002,
+			expectedRWT: testRecordsWithTags002,
+			metrics:     testMetrics002,
+			needModify:  false,
+		},
+		{
+			name:        "03 test different timestamp precision",
+			cli:         cli003,
+			expectedRWT: testRecordsWithTags003,
+			metrics:     testMetrics001,
+			needModify:  false,
+		},
+		{
+			name:        "04 test treat tags as fields",
+			cli:         cli004,
+			expectedRWT: testRecordsWithTags004,
+			metrics:     testMetrics001,
+			needModify:  true,
+		},
+		{
+			name:        "05 test treat tags as device id",
+			cli:         cli005,
+			expectedRWT: testRecordsWithTags005,
+			metrics:     testMetrics001,
+			needModify:  true,
+		},
+	}
+	for _, testCase := range tests {
+		result, err := testCase.cli.convertMetricsToRecordsWithTags(testCase.metrics)
+		require.NoError(t, err)
+		if testCase.needModify {
+			require.NoError(t, testCase.cli.modifyRecordsWithTags(result))
+		}
+		requireRecordsEqual(t, &(testCase.expectedRWT), result, testCase.name)
+	}
+}
+
+// Start a container and do integration test.
+func TestIntegrationInserts(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	container := testutil.Container{
+		Image:        "apache/iotdb:1.13.0",
+		ExposedPorts: []string{testPort},
+		WaitingFor: wait.ForAll(
+			wait.ForListeningPort(nat.Port(testPort)),
+			wait.ForLog("Waiting for connections"),
+		),
+	}
+	err := container.Start()
+	require.NoError(t, err, "failed to start IoTDB container")
+	defer func() {
+		require.NoError(t, container.Terminate(), "terminating IoTDB container failed")
+	}()
+
+	t.Logf("Container Address:%v, ExposedPorts:[%v:%v]", container.Address, container.Ports[testPort], testPort)
+
+	testClient := newTestClient()
```

Comment:
Please instantiate the client here instead of using a wrapper.

Reply:
all right. Thanks for suggestions. But the adapter was not deleted because other functions are still using it.

Reply:
Please remove the function and use `newIoTDB()` of the main file.

Reply:
all right. All default config are in `newIoTDB()` except `Log`. `newTestClient()` are removed.